### PR TITLE
[client] Add Data Component and Data Source entity

### PR DESCRIPTION
**Pull request by RomuDeuxfois**
### Proposed changes

* Add Data Component entity management
* Add Data Source entity management

### Related issues

* [680](https://github.com/OpenCTI-Platform/opencti/issues/680)

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: pycti/utils/opencti_stix2.py [hunk 1]
```diff
@@ -1207,6 +1211,18 @@ def prepare_export(
         result = []
         objects_to_get = []
         relations_to_get = []
+
+        # # Handle prepare_export specific case
+        if entity is not None and "type" in entity:
```
Comment (richard-julien) [inline pycti/utils/opencti_stix2.py:1216]: Need explanation about that :)

File: pycti/api/opencti_api_client.py [hunk 2]
```diff
@@ -497,6 +501,14 @@ def process_multiple_fields(self, data):
         :rtype: dict
         """
 
+        # Handle process_multiple_fields specific case
+        if data is not None and "entity_type" in data:
```
Comment (richard-julien) [inline pycti/api/opencti_api_client.py:505]: Need explanation about that :)

File: pycti/utils/opencti_stix2.py (hunk 1 above)
Comment (RomuDeuxfois) [reply pycti/utils/opencti_stix2.py:1216]: The goal here is to put the entity-specific code for the "prepare_export" and "process_multiple_fields" methods inside the entity file.
But I'm not very comfortable yet with contracts (interface) in python. So, this is the only method I have found so far to do it... Glad to discuss about it

**Comment by Archidoit** [comment]
Search doesn't work for Narratives, Courses of Action and Data Sources (in Techniques)

**Comment by Archidoit** [comment]
Error when filtering a location

**Comment by Archidoit** [comment]
in Regions, the region contouring on the map is always centered on Europe (even if the contouring is in an other continent)

**Comment by Archidoit** [comment]
when we remove alias for a Country, the delimitation of the country on the map disappears

**Comment by Archidoit** [comment]
when we update a Country, it is indicated that the field Description is required but we can still update the country without problem for an other field without filling the Description field.

**Pull request closed without merging**